dashboard/app/models/concept.rb

```diff
@@ -27,15 +27,19 @@ class Concept < ActiveRecord::Base
 
   def self.by_name(name)
     (@@name_cache ||= Concept.all.index_by(&:name))[name].try(:id)
+    Rails.cache.fetch('concepts/names/{name}') do
+      Concept.find_by_name(name).try(:id)
+    end
   end
 
   def self.cached
-    @@all_cache ||= Concept.all
+    Rails.cache.fetch('concepts/all') do
+      Script.all
+    end
   end
 
   def self.expire_cache
-    @@all_cache = nil
-    @@name_cache = nil
+    Rails.cache.delete_if {|key, _| key.start_with?("concepts/")}
   end
```

Comment:
it doesn't look like we actually use this method; should we just remove it?

Reply:
There's a commented out line calling this but I'm guessing that it's not important if we haven't fixed it in 5+ years. :)